javascript/react-js/react-testing-part-one.md

```diff
@@ -100,27 +103,31 @@ Let's test if the button works as intended:
 // App.test.js
 
 import React from "react";
-import { render } from "@testing-library/react";
+import { render, screen } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 import App from "./App";
 
-test("magnificent monkeys renders", () => {
-  const { container } = render(<App />);
-  expect(container).toMatchSnapshot();
-});
+describe("App component", () => {
+  it("renders magnificent monkeys", () => {
+    const { container } = render(<App />);
+    expect(container).toMatchSnapshot();
+  });
```

Comment:
I'm curious why you elected to use restructuring from render here rather than using screen. I know that after our previous discussion you felt that one example destructuring from render should be kept in, but I assumed that was accomplished in lines 58-62?

Reply:
The main reason was to keep the `toMatchSnapshot` assertion here, since it ties pretty well with the next section "What are Snapshots?".

`container` is not a property on the `screen` object it seems, and this document confirms it: https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/prefer-screen-queries.md

It says: 

> DOM Testing Library (and other Testing Library frameworks built on top of it) exports a screen object which has every query (and a debug method). This works better with autocomplete and makes each test a little simpler to write and maintain.
> 
> This rule aims to force writing tests using built-in queries directly from screen object rather than destructuring them from render result. Given the screen component does not expose utility methods such as rerender() or the container property, it is correct to use the render returned value in those scenarios.

One way I could achieve the snapshot test without using `container` was using `document`, but I don't know if that's ideal:

```js
  it("renders magnificent monkeys", () => {
    render(<App />);
    expect(document.body.firstChild).toMatchSnapshot();
  });
```

Otherwise, an alternate way to test would be to just use `screen.getByRole` to assert that the heading is there. That way, we'll not need the `container` at all.

For the time being, I've added a comment to justify render's destructuring in this test:

```js
  it("renders magnificent monkeys", () => {
    // since screen does not have the container property, we'll destructure render to obtain container for this test
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
 ```
 
 Let me know if we should remove/edit this comment. Also let me know if we want to keep this test as is or go with a different approach.